documentation/library_guide/parallel_api/additional_algorithms.rst

```diff
@@ -87,6 +87,22 @@ header.  All algorithms are implemented in the ``oneapi::dpl`` namespace.
     output_keys:   [0,    0,   3,   3,   4,   5]
     output_values: ['c', 'f', 'a', 'e', 'd', 'b']
 
+  .. note::
+
+     ``sort_by_key`` may implement an unstable sort in the future.
+     Use ``stable_sort_by_key`` if the stability is important.
+
+* ``stable_sort_by_key``: performs a stable key-value sort. The algorithm sorts the sequence's keys according to 
+  a comparioson operator. If no comparator is provided, then the elements are compared with ``operator<``.
+  The sequence's values are permutated according to the sorted sequence's keys. The prerequisite for correct
+  behavior is that the size for both keys sequence and values sequence shall be the same.  
+  For example::
+
+    keys:   [3,    5,   0,   4,   3,   0]
+    values: ['a', 'b', 'c', 'd', 'e', 'f']
+    output_keys:   [0,    0,   3,   3,   4,   5]
+    output_values: ['c', 'f', 'a', 'e', 'd', 'b']
+

```

Comment:
In order to  not duplicate the same information, would not it be better to just change `sort_by_key` to `stable_sort_by_key` in the current wording, and add a short section that describes `sort_by_key` as equivalent to the stable one and adds the caution note?

Reply:
That's a good idea. It's done.